### PR TITLE
Virtual machine no longer lazy

### PR DESCRIPTION
**Pull request by Christian-B**
This code builds the whole virtual machine right away.

In other words all chips and processors are filled in rather than left lazy.

### CODE REVIEW

**Comment by rowleya** [comment]
It is worth noting that this slows down the SpallocMaxMachineGenerator algorithm, as that code builds a VM, however it also speeds up the PreAllocateResourcesForExtraMonitorSupport, probably because that used to actually end up filling in the VM.

File: spinn_machine/chip.py [hunk 1]
```diff
@@ -59,27 +69,25 @@ def __init__(self, x, y, processors, router, sdram, nearest_ethernet_x,
...
         self._ip_address = ip_address
         self._virtual = virtual
-        if tag_ids is not None:
-            self._tag_ids = tag_ids
-        else:
-            if self._ip_address is None:
-                self._tag_ids = []
-            else:
-                self._tag_ids = self.IPTAG_IDS
+        self._tag_ids = tag_ids
```
Comment (dkfellows) [inline spinn_machine/chip.py]: This change was wrong.

File: spinn_machine/chip.py [hunk 2]
```diff
@@ -23,10 +25,20 @@ class Chip(object):
...
+            processors[i] = Processor.factory(i)
+        return processors
+
+    DEFAULT_PROCESSORS = default_processors.__func__()
+
     # pylint: disable=too-many-arguments
     def __init__(self, x, y, processors, router, sdram, nearest_ethernet_x,
                  nearest_ethernet_y, ip_address=None, virtual=False,
-                 tag_ids=None):
+                 tag_ids=IPTAG_IDS):
```
Comment (dkfellows) [inline spinn_machine/chip.py]: This change was wrong.

**Comment by rowleya** [comment]
> This PR accidentally undid other work.

I believe this to be fixed in #99.  I haven't merged that yet because of its links to other things, but if that is the only change, it could be merged now.

**Pull request merged**